File: web/src/types/fuelRecord.ts
/* ========================================
   打油记录模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** 打油记录详情（对齐 FuelRecord schema） */
export interface FuelRecord {
  id: number;
  vehicleId: number;
  vehiclePlateNo?: string | null;
  driverUserId?: number | null;
  driverName?: string | null;
  fuelDate: string;
  liters?: number | null;
  amount?: number | null;
  unitPrice?: number | null;
  odometer?: number | null;
  images?: string[] | null;
  remark?: string | null;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** 打油记录列表项（对齐 FuelRecordSummary schema） */
export interface FuelRecordSummary {
  id: number;
  vehicleId: number;
  vehiclePlateNo?: string | null;
  driverName?: string | null;
  fuelDate: string;
  liters?: number | null;
  amount?: number | null;
  imageCount: number;
  createdAt: string;
}

/** 打油记录列表分页响应 */
export interface FuelRecordListResponse {
  items: FuelRecordSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建打油记录请求 */
export interface CreateFuelRecordRequest {
  vehicleId: number;
  driverUserId?: number | null;
  fuelDate: string;
  liters?: number | null;
  amount?: number | null;
  unitPrice?: number | null;
  odometer?: number | null;
  images?: string[];
  remark?: string;
}

/** 更新打油记录请求 */
export interface UpdateFuelRecordRequest {
  vehicleId?: number;
  driverUserId?: number | null;
  fuelDate?: string;
  liters?: number | null;
  amount?: number | null;
  unitPrice?: number | null;
  odometer?: number | null;
  images?: string[];
  remark?: string;
}

/** 打油记录查询参数 */
export interface FuelRecordQueryParams {
  vehicleId?: number;
  fuelDateStart?: string;
  fuelDateEnd?: string;
  page?: number;
  pageSize?: number;
}

File: api/src/main/java/com/aistore/module/fuel/entity/FuelRecord.java
package com.aistore.module.fuel.entity;

import com.aistore.common.handler.JsonbStringListTypeHandler;
import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.FieldStrategy;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.apache.ibatis.type.JdbcType;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 打油（加油）记录实体（表 fuel_record）。
 * images 为 PostgreSQL jsonb（图片 URL 数组），autoResultMap 让 BaseMapper 套用类型处理器；
 * 可空字段标注 updateStrategy=IGNORED，使更新时可写 null（如清空图片/司机）。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(value = "fuel_record", autoResultMap = true)
public class FuelRecord {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long vehicleId;

    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private Long driverUserId;

    private LocalDate fuelDate;

    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private BigDecimal liters;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private BigDecimal amount;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private BigDecimal unitPrice;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private BigDecimal odometer;

    @TableField(typeHandler = JsonbStringListTypeHandler.class, jdbcType = JdbcType.OTHER,
            updateStrategy = FieldStrategy.IGNORED)
    private List<String> images;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}
